### PR TITLE
Add api endpoints

### PR DESCRIPTION
**Pull request by emhamza**
### In this milestone my coding partner and I have created different API end points to my-blog app. Following are the tasks that we have accomplished so far: 🌮 🎉 

- [x] Add routes for API end points
- [x] Create an API endpoint to list all posts for a user.
- [x] Create an API endpoint to list all comments for a user's post.
- [x] Add separate controller so that the Devise authentication will not interfere with API's end points

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,4 +1,9 @@
 Rails.application.routes.draw do
+  namespace :api do
```
Comment (addod19) [inline config/routes.rb]: ### Optional
- [x] It would be nice if we delete the duplicates from the code since we have it already at the bottom of the file

**Review by addod19 — CHANGES_REQUESTED** [review]
## Hi @emhamza and @AbbasSarwar, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- [ ] Correct Git-flow followed 💯 
- [ ] Awesome collaboration 🚀 
- [ ] No linter errors recorded 🎆 
- [ ] Post and comments routes looks great 🚀 
- [ ] Controller actions for post and comment matches 💯 


## Required Changes ♻️
_Check the comments under the review._
- [x] It would be better if we address the README file where we have “cd catalogue-of-my-things” which is not reflecting on the current project hence making the file unprofessional 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (emhamza) [reply config/routes.rb]: @addod19 Very insightful, Thank you so much for pointing out my DRY bug! 😃

**Review by julie-ify — APPROVED** [review]
Hi @emhamza and @AbbasSarwar,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**